FastSimulation/TrackingRecHitProducer/src/PixelTemplateSmearerBase.cc

```diff
@@ -40,28 +49,132 @@ PixelTemplateSmearerBase::PixelTemplateSmearerBase(
     const edm::ParameterSet& config,
     edm::ConsumesCollector& consumesCollector 
 ):
-    TrackingRecHitAlgorithm(name,config,consumesCollector)
+  TrackingRecHitAlgorithm(name,config,consumesCollector)
 {
+    //--- Basic stuff
     mergeHitsOn = config.getParameter<bool>("MergeHitsOn");
-    templateId = config.getParameter<int> ( "templateId" );
+    isBarrel    = config.getParameter<bool> ( "isBarrel" );
+    int detType = (isBarrel) ? 1 : 0;    // 1 for barrel, 0 for forward  (or could we just promote bool into int...?)
+
+    //--- Resolution file names.
+    theBigPixelResolutionFileName     = config.getParameter<string>( "BigPixelResolutionFile" );
+    theEdgePixelResolutionFileName    = config.getParameter<string>( "EdgePixelResolutionFile" );
+    theRegularPixelResolutionFileName = config.getParameter<string>( "RegularPixelResolutionFile" );
+
+
+    //--- Create the resolution histogram objects, which will load the histograms
+    //    and initialize random number generators.
+    //
+    int status = 0;
+    theRegularPixelResolutions = 
+      std::make_shared<PixelResolutionHistograms>( edm::FileInPath( theRegularPixelResolutionFileName ).fullPath().c_str(), "" );
```

Comment:
Isn't this exactly the same problem that was fixed already for some part in the template code earlier in the spring (I'm unable to find the related PRs though), i.e. `edm::FileInPath::fullPath()` returns a new `std::string` object which gets destructed almost immediately, so there is no guarantee that its `char *` buffer (returned by `c_str()`) would be valid at the time of `PixelResolutionHistograms` constructor.

(Could e.g. `PixelResolutionHistograms` take the file name by `const std::string&`?)

Reply:
@makortel 
Possibly.  I was thinking about that yesterday, but I was flying back to the US and couldn't do much web-surfing to check.  The original problem in pixels didn't involve FileInPath, but it did a temporary char* from c_str().   I agree that std::string may be better, so let me try to make this work.

Reply:
@makortel 
OK, I think I fixed it.  The job runs, finds the files, so let's do another round of tests to make sure everything is the same as yesterday.  
@kpedro88 , I'll work on my grid cert next.